src/main/scala/Core/config.scala

```diff
@@ -193,7 +199,32 @@ case class FlexpretConfiguration(
 #define GPO_SIZES       {${gpoPortSizes.mkString(",")}}
 
 #endif
-    """
+
+"""
+  }
+
+  def generateMakeConfig() : String = {
+    return s"""
+#
+# This flexpret core config file is auto-generated, and based on the
+# configuration used to build the flexpret emulator.
+#
+# Do not edit.
+#
+#
+
+HAVE_THREADS := ${threads}
```

Comment:
Couldnt we just skip `HAVE`?

Reply:
The only issue is that the names are already in use in the defconfig.mk makefile, so if both the defconfig.mk and config.mk files are included in the same makefile there will be a clash of variable names. That's why I prefixed them all with "HAVE_". But come to think of it, I don't see a world where these two makfiles should be used together anyway, since defconfig.mk describes the config used to build the FlexPRET while the config.mk describes the live config that was used to build the CPU and is used to configure SW. So it should work to have the name clash.

Come to think of it, the two files should probably be renamed to better signal their two distinct purposes (i.e., defconfig.mk being used to configure CPU build and config.mk being used to configure SW based on the build CPU).

Reply:
Yes I agree, it is a little confusing since we are using Makefiles to compile Chisel and also Makefiles to compile C... change names to make this very clear is a great idea